forem/forem
# Remove autofollowing hiring tag

## What type of PR is this? (check all applicable)


## Description

This PR removes the automatic following of the hiring tag after ticking the "Looking for work" checkbox on a user's profile. This was primarily done for two reasons:

1. It's very DEV specific and dates back to a time where DEV was still exploring other possible futures than Forem.
2. The hiring fields are pretty complex, as they rely on the interaction of two fields (are you looking at all and if yes, is it public?) which doesn't generalize well.

Because of 2. I vote for completely removing the "looking for work publicly" field from all Forems via a data update script once the generalized profile page has been merged (see #11485) as it will no longer serve a purpose. We might even do that as part of the other PR.

I'll leave this PR as draft for now, as we still have some product-related discussion going on in https://github.com/forem/InternalProjectPlanning/issues/366 as to how to notify users who currently have these flags set about the upcoming change.

## Related Tickets & Documents

Closes https://github.com/forem/InternalProjectPlanning/issues/366

## QA Instructions, Screenshots, Recordings

0. Ensure your user doesn't follow the `hiring` tag.
1. Go to `/settings/profile`
2. Check the "Looking for work" checkbox and save your profile:
    ![Screen Shot 2020-12-03 at 12 22 15](https://user-images.githubusercontent.com/47985/100967604-30662600-3562-11eb-8c33-4d00244141d5.png)
3. Your user should still not follow the `hiring` tag.

## Added tests?

- [X] No, and this is why: I'm removing a feature, so I'm also removing the corresponding spec.

## Added to documentation?


## [optional] What gif best describes this PR or how it makes you feel?

![image](https://user-images.githubusercontent.com/47985/100967766-a10d4280-3562-11eb-89e2-267caa2a18ee.png)
